### PR TITLE
Add LiveScript language support

### PR DESCRIPTION
**Pull request by Announcement**
copied coffee-script and modified it for livescript

### CODE REVIEW

**Comment by judofyr** [comment]
Could you add some tests as well?

**Comment by Announcement** [comment]
@judofyr working on that now. What should I do about the README? Tilt Team or Community?

**Comment by Announcement** [comment]
Okay and the tests failing are due to my lack of knowledge on how write a darn test script :100: :-1:

**Comment by judofyr** [comment]
You can put in in `Tilt team` as long it's okay that I bother you if the build happen to break. This is usually caused because the LiveScript gem changes (or maybe the language itself).

No, no. You've done good! That error is quite odd to be honest. It seems like a bug in a combination of ExecJS and Duktape. I'll look into it.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.0.6 (2016-06-03)
```
Comment (judofyr) [inline CHANGELOG.md]: Change this to just `## master`. I'll tag it and add a date when we finally release it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+
+* Add support for LiveScript
```
Comment (judofyr) [inline CHANGELOG.md]: Add `(#286, Announcement)` (or your full name)

File: lib/tilt/livescript.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    @@default_bare = false
+
+    def self.default_bare
+      @@default_bare
+    end
+
+    def self.default_bare=(value)
+      @@default_bare = value
+    end
+
```
Comment (judofyr) [inline lib/tilt/livescript.rb]: This is a feature I don't think is very useful. Libraries that integrate with Tilt should give you a way to configure `options` as you'd like. Also, the LiveScript gem itself has a way to set default options globally.

File: lib/tilt/livescript.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    def prepare
+      if !options.key?(:bare)
+        options[:bare] = self.class.default_bare
+      end
+    end
+
```
Comment (judofyr) [inline lib/tilt/livescript.rb]: For the same reason as above, just make this an empty method:
```ruby
def prepare
end
```

File: test/tilt_livescripttemplate_test.rb [hunk 5]
```diff
@@ -0,0 +1,99 @@
+require 'test_helper'
+require 'tilt'
+
+begin
+  require 'tilt/livescript'
+
+  module LiveScriptTests
+    def self.included(mod)
+      mod.class_eval do
```
Comment (judofyr) [inline test/tilt_livescripttemplate_test.rb]: Unless you want to include LiveScriptTests in multiple test classes, you might as well write the code below directly into the test class:

```
class LiveScriptTemplateTest < Minitest::Test
  setup do
    …
  end

  test "bare is disabled by default" do
    …
  end
```

**Comment by judofyr** [comment]
Ignore the test failures for now (or, I assume it works locally on your machine?). I've added some comments. Nothing very big. You can fix them yourself if you'd like, or I can fix it some time next week. Let me know if anything is unclear.

**Comment by judofyr** [comment]
(I could probably quite easily fix them myself and merge it right away, I think it's nice that you'll have a say at what finally ends up in the gem.)

**Comment by judofyr** [comment]
I know what the issue is. It was fixed in Duktape v1.3.0.5. I've increased the dependency in the master branch. Could you try to merge master into this branch and see if it works then?

**Comment by judofyr** [comment]
Also: Running tests locally are much easier if you do: `bundle install --without secondary` first. Then most of the tests will pass (`bundle exec rake test`).

File: lib/tilt/livescript.rb (hunk 3 above)
Comment (Announcement) [inline lib/tilt/livescript.rb]: So just to be clear, remove self.default_bare?

**Comment by Announcement** [comment]
Don't know if i'm reading the messy tests suite right, but it all looks like it passed ~ hooray!
really wish rake was pretty like javascript's mocha / chai. - guess I know what I'll be contributing to next ;)

**Comment by Announcement** [comment]
Don't know if i'm reading the messy tests suite right, but it all looks like it passed ~ hooray!
really wish rake was pretty like javascript's mocha / chai. - guess I know what I'll be contributing to next ;)

**Comment by Announcement** [comment]
btw can we look into updating dependencies. i don't like the git badge saying we're out dated lol

**Comment by judofyr** [comment]
In some version of Rake they turned on warnings by default in the test cases. As you can see, the original test case was *not* written warning-free. The proper fix is of course to fix all the warnings, but a short-fix is not to enable them. I've just ignored it for now.

**Comment by judofyr** [comment]
After removing the support for `default_bare`, we might as well rewrite the test case to just this:

```ruby
require 'test_helper'
require 'tilt'

begin
  require 'tilt/livescript'

  class LiveScriptTemplateTest < Minitest::Test
    setup do
      @code_without_variables = "puts 'Hello, World!'\n"
      @renderer = Tilt::LiveScriptTemplate
    end

    test "compiles and evaluates the template on #render" do
      template = @renderer.new { |t| @code_without_variables }
      assert_match "puts('Hello, World!');", template.render
    end

    test "can be rendered more than once" do
      template = @renderer.new { |t| @code_without_variables }
      3.times { assert_match "puts('Hello, World!');", template.render }
    end

    test "supports bare-option" do
      template = @renderer.new(:bare => false) { |t| @code_without_variables }
      assert_match "function", template.render

      template = @renderer.new(:bare => true) { |t| @code_without_variables }
      refute_match "function", template.render
    end

    test "is registered for '.ls' files" do
      assert_equal @renderer, Tilt['test.ls']
    end
  end
rescue LoadError => boom
  warn "Tilt::LiveScriptTemplate (disabled)"
end
```

**Comment by judofyr** [comment]
I'll accept the pull request if we change the test case to the code I just posted. I already have it working locally, so if you're fine with that I can merge it back to master.

**Pull request merged**